src/com/owncloud/android/ui/activity/PatternLockActivity.java

```diff
@@ -0,0 +1,297 @@
+            setCancelButtonEnabled(false);
+        } else if (ACTION_REQUEST_WITH_RESULT.equals(getIntent().getAction())) {
+            if (savedInstanceState != null) {
+                mPatternPresent = savedInstanceState.getBoolean(KEY_CONFIRMING_PATTERN);
+                mPatternValue = savedInstanceState.getString(KEY_PATTERN_STRING);
+                mPatternHeaderViewText = savedInstanceState.getString(PATTERN_HEADER_VIEW_TEXT);
+                mPatternExpViewState = savedInstanceState.getBoolean(PATTERN_EXP_VIEW_STATE);
+                mCount = savedInstanceState.getInt(COUNT_VALUE);
+            }
+            if (mPatternPresent) {
+                mPatternHeader.setText(mPatternHeaderViewText);
+                if (!mPatternExpViewState) {
+                    mPatternExplanation.setVisibility(View.INVISIBLE);
+                }
+                checkPattern();
+            } else {
+                mPatternHeader.setText(R.string.pattern_configure_pattern);
+                mPatternExplanation.setVisibility(View.VISIBLE);
+                setCancelButtonEnabled(true);
+            }
+        } else if (ACTION_CHECK_WITH_RESULT.equals(getIntent().getAction())) {
+            mPatternHeader.setText(R.string.pattern_remove_pattern);
+            mPatternExplanation.setVisibility(View.VISIBLE);
+            setCancelButtonEnabled(true);
+        } else {
+            throw new IllegalArgumentException(R.string.illegal_argument_exception_message + " " +
+                    TAG);
+        }
+        setPatternListener();
+    }
+
+    /**
+     * Binds the appropiate listener to the pattern view.
+     */
+    protected void setPatternListener() {
+        mPatternLockView.addPatternLockListener(new PatternLockViewListener() {
+            @Override
+            public void onStarted() {
+                Log_OC.d(TAG, "Pattern Drawing Started");
+            }
+
+            @Override
+            public void onProgress(List<PatternLockView.Dot> list) {
+                Log_OC.d(TAG, "Pattern Progress " +
+                        PatternLockUtils.patternToString(mPatternLockView, list));
+            }
+
+            /**
+
+             */
+            @Override
+            public void onComplete(List<PatternLockView.Dot> list) {
+                if (ACTION_REQUEST_WITH_RESULT.equals(getIntent().getAction())) {
+                    /**
+                     * This block gets executed when the pattern has to be set.
+                     * count variable holds the number of time the pattern has been input.
+                     * if the value of count is two then the pattern input first (which is stored in patternValue variable) is compared with
+                     * the pattern value input the second time (which is stored in newPatternValue) if both the variables hold the same value
+                     * then the pattern is set.
+                     */
+                    if (mCount == 0) {
```

Comment:
Do you really need this counter? You could check the second time whether mPatternValue is not empty and set the input to mNewPatternValue in that case.

Reply:
@davigonz Yes that can be done. I will do that. Thanks.